### PR TITLE
No issue: Add device name to account page

### PR DESCRIPTION
**Pull request by jonalmeida**
This is taken from r-b with a bit of polish for Fenix.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt [hunk 1]
```diff
@@ -94,6 +112,24 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
...
+        return Preference.OnPreferenceChangeListener { _, newValue ->
+            // Optimistically set the device name to what user requested.
+            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
+            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
+            preferenceDeviceName?.summary = newValue as String
+
+            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
+            CoroutineScope(Dispatchers.IO).launch {
+                accountManager.authenticatedAccount()?.let {
+                    it.deviceConstellation().setDeviceNameAsync(newValue)
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt]: Can you try/catch this to make sure we don't propagate errors here? re-throw FxaPanicException, and simply log the rest of the FxA error types for now.

File: app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt [hunk 2]
```diff
@@ -94,6 +114,28 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
...
+            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
+            preferenceDeviceName?.summary = newValue as String
+
+            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
+            CoroutineScope(Dispatchers.IO).launch {
+                try {
+                    accountManager.authenticatedAccount()?.let {
+                        it.deviceConstellation().setDeviceNameAsync(newValue)
+                    }
+                } catch (e: FxaException) {
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt]: Don't swallow panics here.

```
} catch (e: FxaPanicException) {
    throw e
} catch (e: FxaException) {
    log...
}
```

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt]: Thanks for clarifying my misunderstanding!

**Comment by grigoryk** [comment]
Looks good now, thanks @jonalmeida . I'd still like to see those screenshots ;) But this should be good to land.

**Comment by jonalmeida** [comment]
@grigoryk sorry about that, here they are!

Screenshot_20190524-170732
Screenshot_20190524-170738

**Comment by grigoryk** [comment]
:ship:

**Pull request merged**